src/test/java/org/broadinstitute/hellbender/tools/pon/coverage/pca/HDF5PCACoveragePoNUnitTest.java

```diff
@@ -0,0 +1,232 @@
+        Assert.assertNotNull(actual);
+        Assert.assertEquals(actual.getRowDimension(), targets.size());
+        Assert.assertTrue(actual.getColumnDimension() <= samples.size());
+        final RealMatrix expected = readDoubleMatrix(TEST_PON_REDUCED_PON);
+        assertEqualMatrix(actual, expected);
+    }
+
+    @Test(dependsOnMethods = {"testTargetNameReading","testLogNormalizedSampleNameReading"})
+    public void testReducedPoNPInvMatrixReading() throws IOException {
+        final HDF5File reader = new HDF5File(TEST_PON);
+        final PCACoveragePoN pon = new HDF5PCACoveragePoN(reader);
+        final List<String> targets = pon.getTargetNames();
+        final List<String> samples = pon.getPanelSampleNames();
+        final RealMatrix actual = pon.getReducedPanelPInverseCounts();
+        Assert.assertNotNull(actual);
+        Assert.assertTrue(actual.getRowDimension() <= samples.size());
+        Assert.assertEquals(actual.getColumnDimension(), targets.size());
+        final RealMatrix expected = readDoubleMatrix(TEST_PON_REDUCED_PON_PINV);
+        assertEqualMatrix(actual, expected);
+    }
+
+    private void assertEqualMatrix(RealMatrix actual, RealMatrix expected) {
+        Assert.assertEquals(actual.getRowDimension(), expected.getRowDimension());
+        Assert.assertEquals(actual.getColumnDimension(), expected.getColumnDimension());
+        for (int row = 0; row < expected.getRowDimension(); row++) {
+            for (int column = 0; column < expected.getColumnDimension(); column++) {
+                final double actualValue = actual.getEntry(row,column);
+                final double expectedValue = expected.getEntry(row,column);
+                final double epsilon = Math.min(Math.abs(actualValue),Math.abs(expectedValue)) * 0.0001;
+                Assert.assertEquals(actualValue,expectedValue,epsilon);
+            }
+        }
+    }
+
+    /**
+     * Reads the lines of a file into an string array.
+     * @param file the input file.
+     * @return never {@code null}, a list with as many elements as lines in {@code file}.
+     * @throws IOException if some IO exception occurred.
+     */
+    private List<String> readLines(final File file) throws IOException {
+
+        final BufferedReader reader = new BufferedReader(new FileReader(file));
+        final ArrayList<String> result = new ArrayList<>(1000);
+        String line;
+        while ((line = reader.readLine()) != null) {
+            result.add(line);
+        }
+        return result;
+    }
+
+    /**
+     * Reads the lines of a file into a double list.
+     * @param file the input file.
+     * @return never {@code null}, a list with as many elements as lines in {@code file}.
+     * @throws IOException if some IO exception occurred.
+     * @throws NumberFormatException if some of the input lines in {@code file} cannot be converted into a {@code double}.
+     */
+    private List<Double> readDoubleLines(final File file) throws IOException {
```

Comment:
eh...seems like a method just for test files, is that the appropriate place?  i think all our production code should go through TableReader/TableWriter or something more robust, but other tests could make use of these methods.  punting for now